include/flatbuffers/flatbuffers.h

```diff
@@ -836,15 +849,20 @@ class FlatBufferBuilder FLATBUFFERS_FINAL_CLASS {
     return CreateVectorOfStructs(v.data(), v.size());
   }
 
+  template<typename T>
```

Comment:
Using local types as templates is only supported post gcc 4.5 (https://gcc.gnu.org/projects/cxx0x.html), so in order to support 4.4, this will have to be defined outside of the function. However, changed this to be a struct.